jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/BuildImageStep.java

```diff
@@ -96,8 +101,16 @@
           buildAndCacheApplicationLayerSteps) {
         imageBuilder.addLayer(NonBlockingSteps.get(buildAndCacheApplicationLayerStep));
       }
+
+      // Use environment from base image
+      imageBuilder.setEnvironment(
+          NonBlockingSteps.get(pullBaseImageStep).getBaseImage().getEnvironment());
+      if (buildConfiguration.getEnvironment() != null) {
+        for (Map.Entry<String, String> entry : buildConfiguration.getEnvironment().entrySet()) {
+          imageBuilder.setEnvironmentVariable(entry.getKey(), entry.getValue());
+        }
```

Comment:
I wonder if this could be simplified by instead having just a `imageBuilder.addEnvironment()`:
```diff
+      // Initialize to environment from base image
+      imageBuilder.addEnvironment(
+          NonBlockingSteps.get(pullBaseImageStep).getBaseImage().getEnvironment());
+      // override with configured environment
+      if (buildConfiguration.getEnvironment() != null) {
+         imageBuilder.addEnvironment(buildConfiguration.getEnvironment());
+      }